component/status/status.go

```diff
@@ -0,0 +1,138 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status // import "go.opentelemetry.io/collector/component/status"
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+// EventType represents an enumeration of status event types
+type EventType int
+
+const (
+	// EventOK indicates the producer of a status event is functioning normally
+	EventOK EventType = iota
+	// EventError indicates the producer of a status event is in an error state
+	EventError
```

Comment:
Why not to extend this list with: OK, RECOVERABLE_ERROR, PERMANENT_ERROR, FATAL_ERROR **vs** use the combination of EventError + componenterror type?

Reply:
I made this change and got rid of the newly introduced error types.